File: design-patterns/ts/mediator/general/colleague2.ts
// colleague2.ts

import Colleague from "./colleague";
import Mediator from "./mediator";


class Colleague2 extends Colleague {
    constructor(mediator: Mediator) {
        super(mediator);
    }

    sendMessage(colleague: Colleague, msg: string): void {
        this.mediator.sendMessage(colleague, msg);
    }

    receiveMessage(msg: string): void {
        console.log("Message received in Colleague2: " + msg);
    }
}

export default Colleague2;

File: design-patterns/go/mediator/general/colleague2.go
// colleague2.go

package main

import "fmt"


type Colleague2 struct {
	*Colleague
}

func NewColleague2(mediator *Mediator) (colleague2 *Colleague2) {
	colleague2 = &Colleague2{}
	colleague2.Colleague = NewColleague(mediator)
	return
}

func (colleague2 *Colleague2) ReceiveMessage(msg string) {
	fmt.Printf("Message received in Colleague2: %s\n",  msg)
}

func (colleague2 *Colleague2) SendMessage(colleague IColleague, msg string) {
	colleague2.mediator.SendMessage(colleague, msg)
}